src/lib/optimizer/strategy/semi_join_reduction_rule.hpp

```diff
@@ -45,11 +45,12 @@ class PredicateNode;
 
 class SemiJoinReductionRule : public AbstractRule {
  public:
-  void apply_to(const std::shared_ptr<AbstractLQPNode>& root) const override;
-
   // Defines the minimum selectivity for a semi join reduction to be added. For a candidate location in the LQP with an
   // input cardinality `i`, the output cardinality of the semi join has to be lower than `i * MINIMUM_SELECTIVITY`.
   constexpr static auto MINIMUM_SELECTIVITY = .25;
+
+ protected:
+  void _apply_to(const std::shared_ptr<AbstractLQPNode>& lqp_root) const override;
```

Comment:
Could you grep for "memoiz" and update the comments accordingly?

Reply:
Yep, I missed the Doc in stored_table_column_alignment_rule.hpp